src/pydvl/utils/types.py

```diff
@@ -71,3 +76,72 @@ def items(self):
     setattr(cls, "items", items)
 
     return cls
+
+
+def maybe_add_argument(fun: Callable, new_arg: str):
+    """Wraps a function to accept the given keyword parameter if it doesn't
+    already.
+
+    If `fun` already takes a keyword parameter of name `new_arg`, then it is
+    returned as is. Otherwise, a wrapper is returned which merely ignores the
+    argument.
+
+    :param fun: The function to wrap
+    :param new_arg: The name of the argument that the new function will accept
+        (and ignore).
+    :return: A new function accepting one more keyword argument.
+    """
+    params = inspect.signature(fun).parameters
+    if new_arg in params.keys():
+        return fun
+
+    def wrapper(*args, **kwargs):
```

Comment:
Maybe decorating this with `functools.wraps` would be a good idea, even though it's only used internally.

Reply:
Thought about it, but didn't want to think about that for no particular benefit. Why do you think we should add it?

Reply:
Forget it, it's easier to just add it 😅

Reply:
It turns out it isn't. Something breaks if I do (pickling of parallel jobs, don't ask, I'm just going to ignore this 😇 )